core/src/main/java/nl/nn/adapterframework/jta/btm/BtmJtaTransactionManager.java

```diff
@@ -37,6 +37,7 @@ protected UserTransaction retrieveUserTransaction() throws TransactionSystemExce
 	protected BitronixTransactionManager createTransactionManager() {
 		Configuration configuration = TransactionManagerServices.getConfiguration();
 		configuration.setServerId(getUid());
+		configuration.setJdbcProxyFactoryClass("bitronix.tm.resource.jdbc.proxy.JdbcJavaProxyFactory");
```

Comment:
Forcing the use of this JdbcProxyFactoryClass always will fix BTM with newer JDKs. Otherwise BTM will try to use by preference a Javassist version, or a CGLIB version, neither of them working on JDK17 and newer.